apps/wallet/src/shared/deepBook/context.tsx

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { useActiveAccount } from '_app/hooks/useActiveAccount';
+import { useCreateAccount, useMarketAccountCap } from '_hooks';
+import { useSuiClient } from '@mysten/dapp-kit';
+import { DeepBookClient } from '@mysten/deepbook';
+import { createContext, useContext, useEffect, useMemo, type ReactNode } from 'react';
+
+type DeepBookContextProps = {
+	client: DeepBookClient;
+};
+
+const DeepBookContext = createContext<DeepBookContextProps | null>(null);
+
+interface DeepBookContextProviderProps {
+	children: ReactNode;
+}
+
+export function useDeepBookContext() {
+	const context = useContext(DeepBookContext);
+	if (!context) {
+		throw new Error('useDeepBookContext must be used within a DeepBookContextProvider');
+	}
+	return context;
+}
+
+export function useDeepBookClient() {
+	return useDeepBookContext().client;
+}
+
+export function DeepBookContextProvider({ children }: DeepBookContextProviderProps) {
+	const suiClient = useSuiClient();
+	const activeAccount = useActiveAccount();
+
+	const { data, isLoading, refetch } = useMarketAccountCap(activeAccount?.address);
```

Comment:
I don't love that we force fetch tis as part of every app load now, especially seeing as we only need it in some places. Can we either defer this (possible by making the client construction lazy (can still use the context to share)), or move the context just into the pages that need it?